### PR TITLE
Fix for worker sending wrong executionID

### PR DESCRIPTION
**Pull request by yashsinghcodes**
This fix is for worker sending in the wrong executionID to the backend. This cause made it so some of the execution will never be finished.

The solution is quite simple I just removed the dependency onto the variable which was somehow was overwritten in the main memory.

### CODE REVIEW

File: functions/onprem/worker/worker.go [hunk 1]
```diff
@@ -3792,6 +3784,8 @@ func checkUnfinished(resp http.ResponseWriter, request *http.Request, execReques
 	exec, err := shuffle.GetWorkflowExecution(ctx, execRequest.ExecutionId)
 	log.Printf("[DEBUG][%s] Rechecking execution and it's status to send to backend IF the status is EXECUTING (%s - %d/%d finished)", execRequest.ExecutionId, exec.Status, len(exec.Results), len(exec.Workflow.Actions))
 	if err != nil {
+		log.Printf("[ERROR][%s] Got error: %s", execRequest.ExecutionId, err)
+		handleRunExecution(resp, request)
```
Comment (0x0elliot) [inline functions/onprem/worker/worker.go:3788]: not required here

File: functions/onprem/worker/worker.go [hunk 2]
```diff
@@ -3880,13 +3875,16 @@ func handleRunExecution(resp http.ResponseWriter, request *http.Request) {
...
 	var workflowExecution shuffle.WorkflowExecution
-	data = fmt.Sprintf(`{"execution_id": "%s", "authorization": "%s"}`, execRequest.ExecutionId, execRequest.Authorization)
 	streamResultUrl := fmt.Sprintf("%s/api/v1/streams/results", baseUrl)
 	req, err := http.NewRequest(
 		"POST",
 		streamResultUrl,
-		bytes.NewBuffer([]byte(data)),
+		bytes.NewBuffer([]byte(fmt.Sprintf(`{"execution_id": "%s", "authorization": "%s"}`, execRequest.ExecutionId, execRequest.Authorization))),
 	)
+	if err != nil {
```
Comment (0x0elliot) [inline functions/onprem/worker/worker.go:3884]: return an error here in the HTTP request

**Review by 0x0elliot — CHANGES_REQUESTED** [review]

File: functions/onprem/worker/worker.go (hunk 2 above)
Comment (yashsinghcodes) [reply functions/onprem/worker/worker.go:3879]: done

File: functions/onprem/worker/worker.go (hunk 1 above)
Comment (yashsinghcodes) [reply functions/onprem/worker/worker.go]: done

File: functions/onprem/worker/worker.go [hunk 3]
```diff
@@ -3880,13 +3870,18 @@ func handleRunExecution(resp http.ResponseWriter, request *http.Request) {
 	}
 
 	var workflowExecution shuffle.WorkflowExecution
-	data = fmt.Sprintf(`{"execution_id": "%s", "authorization": "%s"}`, execRequest.ExecutionId, execRequest.Authorization)
 	streamResultUrl := fmt.Sprintf("%s/api/v1/streams/results", baseUrl)
 	req, err := http.NewRequest(
 		"POST",
 		streamResultUrl,
-		bytes.NewBuffer([]byte(data)),
+		bytes.NewBuffer([]byte(fmt.Sprintf(`{"execution_id": "%s", "authorization": "%s"}`, execRequest.ExecutionId, execRequest.Authorization))),
```
Comment (0x0elliot) [inline functions/onprem/worker/worker.go:3877]: i still don't know why this helps :)

mind elaborating? is `data` being updated somewhere that i am missing?

**Review by 0x0elliot — CHANGES_REQUESTED** [review]

**Pull request merged**